src/main/java/org/folio/service/orders/CompositePoLineValidationService.java

```diff
@@ -345,27 +345,20 @@ private void validateReceivingWorkflowForBindary(CompositePoLine poLine, List<Er
     }
   }
 
-  public Future<Void> validateUserUnaffiliatedLocationUpdates(String poLineId, List<Location> updatedPoLineLocations,
-                                                              List<Location> storedPoLineLocations, RequestContext requestContext) {
+  public Future<Void> validateUserUnaffiliatedLocationUpdates(String poLineId, List<Location> locations, RequestContext requestContext) {
     return getUserTenantsIfNeeded(requestContext)
       .compose(userTenants -> {
         if (CollectionUtils.isEmpty(userTenants)) {
           logger.info("validateUserUnaffiliatedLocationUpdates:: User tenants is empty");
           return Future.succeededFuture();
         }
-        var storageUnaffiliatedLocations = extractUnaffiliatedLocations(storedPoLineLocations, userTenants);
-        var updatedUnaffiliatedLocations = extractUnaffiliatedLocations(updatedPoLineLocations, userTenants);
-        logger.info("validateUserUnaffiliatedLocationUpdates:: Found unaffiliated POL location tenant ids, poLineId: '{}', stored: '{}', updated: '{}'",
-          poLineId,
-          storageUnaffiliatedLocations.stream().map(Location::getTenantId).distinct().toList(),
-          updatedUnaffiliatedLocations.stream().map(Location::getTenantId).distinct().toList());
-        if (!SetUtils.isEqualSet(storageUnaffiliatedLocations, updatedUnaffiliatedLocations)) {
-          logger.info("validateUserUnaffiliatedLocationUpdates:: User is not affiliated with all locations on the POL, poLineId: '{}'",
-            poLineId);
-          return Future.failedFuture(new HttpException(422, ErrorCodes.LOCATION_UPDATE_WITHOUT_AFFILIATION));
+        var uniqueUnAffiliatedLocations = extractUnaffiliatedLocations(locations, userTenants).stream()
+          .map(Location::getTenantId).distinct().toList();
+        if (CollectionUtils.isNotEmpty(uniqueUnAffiliatedLocations)) {
+          logger.info("validateUserUnaffiliatedLocationUpdates:: User has unaffiliated locations on the POL, poLineId: {}, unique locations: {}", poLineId, uniqueUnAffiliatedLocations);
+          return Future.failedFuture(new HttpException(HttpStatus.SC_UNPROCESSABLE_ENTITY, ErrorCodes.LOCATION_UPDATE_WITHOUT_AFFILIATION));
```

Comment:
also lets add as a param of error poLine Id and uniqueAfflicatedLocations 
var params = List.of(new Parameter().withKey("poLineId......
errors.withParams..

Reply:
This PR is a bugfix, probably it's not safe to change the message on the fly